contrib/coredns/policy/policy.go

```diff
@@ -150,7 +174,47 @@ func (p *PolicyMiddleware) ServeDNS(ctx context.Context, w dns.ResponseWriter, r
 	}
 
 	if response.Permit {
-		return middleware.NextOrFailure(p.Name(), p.Next, ctx, w, r)
+		lr := new(NewLocalResponseWriter)
+		status,err := middleware.NextOrFailure(p.Name(), p.Next, ctx, lr, r)
+		if err != nil {
+			return status, err
+		}
+
+		state := request.Request{W: lr, Req: lr.Msg}
+
+		rr := lr.Msg.Answer[0]
+		switch state.Family() {
+		case 1:
+			attrs = append(attrs, &pb.Attribute{Id: "address", Type: "address", Value: rr.(*dns.A).A.String()})
```

Comment:
Does this type cast work here? Probably you should do a type assertion using the value, ok idiom. In that case you could get rid of the `state` variable since you won't need to check `Family()`. Something like:

```
if a, ok := rr.(*dns.A); ok {
    attrs = append(attrs, &pb.Attribute{Id: "address", Type: "address", Value: a.A.String()})
} else if a, ok := rr.(*dns.AAAA); ok {
    attrs = append(attrs, &pb.Attribute{Id: "address", Type: "address", Value: a.AAAA.String()})
}
```